Automattic/jetpack
# Top Posts & Pages Endpoint: Remove Unnecessary Parameters

Follow up to #34479

## Proposed changes:
It just occurred to me that I messed up a little in #34479 when I refactored the endpoint because it's not necessary to maintain some of the parameters anymore. There's also no issue with now adding a `permission_callback` since the block uses the Helper function directly when rendering it on the front end. 

The endpoint already returns a few extra posts to handle configuring content types within the Block Editor. It's not possible to filter these out from the API request to the Stats endpoint, so this approach prevents the need to make another API request whenever the user changes content types. 

## Jetpack product discussion


N/A.

## Does this pull request change what data or activity we track or use?


No, this is a Janitorial PR.

## Testing instructions:






Enable Beta blocks and insert a Top Posts and Pages block. If they load in both the Block Editor and when previewed on the site, then everything should be fine. You can also check that `$is_rendering_block` works by disabling some of the content types to display (eg. display pages instead of posts).

<img width="683" alt="Screenshot 2023-12-13 at 13 58 41" src="https://github.com/Automattic/jetpack/assets/43215253/3b1dfdba-53d4-43bf-b07b-eced141b5bad">
